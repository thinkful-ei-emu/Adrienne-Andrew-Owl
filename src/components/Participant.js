import React from 'react';

class Participant extends React.Component {
  render() {
    return (
      <div className='participant'>
        <img src={this.props.avatar} alt='Avatar Here but no Avatar, sucks'></img>
        <p>{this.props.name}</p>
        <p>{this.props.inSession ? 'In session' : 'Offline'}</p>
        <p>{this.props.onStage ? 'On Stage' : 'In Session'}</p>
      </div>
    )
  }
}

export default Participant;