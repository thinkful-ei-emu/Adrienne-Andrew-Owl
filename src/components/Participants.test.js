import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Participant from './Participant';
import participants from '../data/participantData';

// smoke test; failing time
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Participant
    key={0}
    name=''
    avatar=''
    inSession={true}
    onStage={false}
    />, div);
  ReactDOM.unmountComponentAtNode(div);
});

// screenshot test;
it('renders this UI as expected', () => {
  const p = participants[0];
  const appTest = renderer.create(<Participant
    key={p.id}
    name={p.name}
    avatar={p.avatar}
    inSession={p.inSession}
    onStage={p.onStage}/>).toJSON();
  // Check whether it matches the previous snapshot
  // Stored in __snapshots__/App.test.js.snap
  expect(appTest).toMatchSnapshot(); 
});