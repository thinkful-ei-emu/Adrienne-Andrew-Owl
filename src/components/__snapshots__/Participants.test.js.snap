// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders this UI as expected 1`] = `
<div
  className="participant"
>
  <img
    alt="Avatar Here but no Avatar, sucks"
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
  />
  <p>
    Koren Templeton
  </p>
  <p>
    
  </p>
  <p>
    On Stage
  </p>
</div>
`;
